supabase/functions/_script/backup_and_cleanup.ts

```diff
@@ -0,0 +1,97 @@
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@^2.2.3';
+import 'https://deno.land/x/dotenv/load.ts';
+import { r2 } from '../_utils/r2';
+import type { Database } from '../_utils/supabase.types.ts';
+import * as fs from 'https://deno.land/std/fs/mod.ts';
+import * as path from 'https://deno.land/std/path/mod.ts';
+
+const supabaseUrl = Deno.env.get('SUPABASE_URL');
+const supabaseServiceRoleKey = Deno.env.get('SUPABASE_SERVICE_ROLE_KEY');
+const backupFolder = 'backup';
+
+if (!supabaseUrl || !supabaseServiceRoleKey) {
+  console.error('Error: SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY environment variables are required.');
+  Deno.exit(1);
+}
+
+function useSupabase() {
+  const options = {
+    auth: {
+      autoRefreshToken: true,
+      persistSession: true,
+      detectSessionInUrl: false,
+    },
+  };
+  return createClient<Database>(supabaseUrl, supabaseServiceRoleKey, options);
+}
+
+async function backupAndDeleteOldEntries() {
+  const currentDate = new Date();
+  const backupFilenames: string[] = [];
+  const recordsToDelete: Database['public']['Tables']['app_versions']['Row'][] = [];
+
+  try {
+    if (!await fs.exists(backupFolder)) {
+      await fs.ensureDir(backupFolder);
+    }
+
+    const supabase = useSupabase();
+    const daysToKeep = 30;
+    const pageSize = 1000;
+    let offset = 0;
+    let hasMoreData = true;
+
+    while (hasMoreData) {
+      const cutoffDate = new Date(currentDate.getTime() - daysToKeep * 24 * 60 * 60 * 1000);
+
+      const { data: oldEntries, error: queryError } = await supabase
+        .from<Database['public']['Tables']['app_versions']['Row']>('app_versions')
+        .select()
+        .lt('updated_at', cutoffDate.toISOString())
+        .range(offset, offset + pageSize - 1);
+
+      if (queryError) {
+        console.error('Error querying the Supabase table:', queryError.message);
+        break;
```

Comment:
Do we want to just skip this iteration? and not stop the processing for all the entries?

Reply:
So in here if one of the processing fails the loop doesn't proceed forward, correct?

Reply:
Do we need to skip the current iteration with an error but continue processing the remaining records?

Reply:
I think @riderx is the one to make that call here, I'd say that you can still process the rest but yeah not sure if even those would be succesful.

Reply:
As said earlier, I'm not sure the Supabase SDK is the right approach for this. Pure SQL could be better.
For the answer if anything fail the script should stop

Reply:
Awesome, @ayusham001 you can resolve this but shall edit to use SQL query instead.